test/adapters/flexibleLeverageStrategyAdapter.spec.ts

```diff
@@ -1501,6 +1620,65 @@ describe("FlexibleLeverageStrategyAdapter", () => {
         });
       });
 
+      context("when using two exchanges", async () => {
+        let subjectExchangeToUse: string;
+
+        cacheBeforeEach(async () => {
+          const newExchangeSettings: ExchangeSettings = {
+            twapMaxTradeSize: ether(2),
+            incentivizedTwapMaxTradeSize: exchangeSettings.incentivizedTwapMaxTradeSize,
+            exchangeLastTradeTimestamp: exchangeSettings.exchangeLastTradeTimestamp,
+            leverExchangeData: EMPTY_BYTES,
+            deleverExchangeData: EMPTY_BYTES,
+          };
+
+          await flexibleLeverageStrategyAdapter.updateEnabledExchange(exchangeName, newExchangeSettings);
+          await flexibleLeverageStrategyAdapter.addEnabledExchange(exchangeName2, newExchangeSettings);
+
+          await chainlinkCollateralPriceMock.setPrice(BigNumber.from(850).mul(10 ** 8));
+          await setV2Setup.usdc.transfer(tradeAdapterMock.address, usdc(100));
+          await setV2Setup.usdc.transfer(tradeAdapterMock2.address, usdc(100));
+        });
+
+        beforeEach(() => {
+          subjectCaller = owner;
+          subjectExchangeToUse = exchangeName;
+        });
+
+        async function subject(): Promise<any> {
+          return flexibleLeverageStrategyAdapter.connect(subjectCaller.wallet).rebalance(subjectExchangeToUse);
+        }
+
+        describe("when leverage ratio is above max and it drops further between rebalances", async () => {
+          it("should set the global and exchange timestamps correctly", async () => {
+            await subject();
+            const timestamp1 = await getLastBlockTimestamp();
+
+            subjectExchangeToUse = exchangeName2;
+            await chainlinkCollateralPriceMock.setPrice(BigNumber.from(800).mul(10 ** 8));
+
+            await subject();
+            const timestamp2 = await getLastBlockTimestamp();
+
+            expect(await flexibleLeverageStrategyAdapter.globalLastTradeTimestamp()).to.eq(timestamp2);
+            expect((await flexibleLeverageStrategyAdapter.getExchangeSettings(exchangeName)).exchangeLastTradeTimestamp).to.eq(timestamp1);
+            expect((await flexibleLeverageStrategyAdapter.getExchangeSettings(exchangeName2)).exchangeLastTradeTimestamp).to.eq(timestamp2);
+          });
+        });
+
+        describe("when leverage ratio is above max and rebalance is called twice", async () => {
+
+          beforeEach(async () => {
+            await subject();
+            subjectExchangeToUse = exchangeName2;
+          });
+
+          it("should revert", async () => {
+            await expect(subject()).to.be.revertedWith("Cooldown not elapsed or not valid leverage ratio");
+          });
+        });
+      });
+
```

Comment:
One more test case just for our sanity. When leverage ratio is within bounds and rebalance is called before rebalance interval has elapsed